pkg/apis/storagemigration/validation/validation.go

```diff
@@ -17,28 +17,34 @@ limitations under the License.
 package validation
 
 import (
-	"fmt"
-	"regexp"
-	"strconv"
+	"strings"
 
-	"k8s.io/apimachinery/pkg/util/sets"
-	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/resourceversion"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/kubernetes/pkg/apis/storagemigration"
 
-	corev1 "k8s.io/api/core/v1"
+	metaconditions "k8s.io/apimachinery/pkg/api/meta"
 	apimachineryvalidation "k8s.io/apimachinery/pkg/api/validation"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	metav1validation "k8s.io/apimachinery/pkg/apis/meta/v1/validation"
+	utilvalidation "k8s.io/apimachinery/pkg/util/validation"
 	apivalidation "k8s.io/kubernetes/pkg/apis/core/validation"
 )
 
 func ValidateStorageVersionMigration(svm *storagemigration.StorageVersionMigration) field.ErrorList {
 	allErrs := field.ErrorList{}
 	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&svm.ObjectMeta, false, apimachineryvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))...)
 
-	allErrs = checkAndAppendError(allErrs, field.NewPath("spec", "resource", "resource"), svm.Spec.Resource.Resource, "resource is required")
-	allErrs = checkAndAppendError(allErrs, field.NewPath("spec", "resource", "version"), svm.Spec.Resource.Version, "version is required")
+	// Same validations as APIService, Group must be a DNS1123 Subdomain and Resource must be DNS1035
+	if errs := utilvalidation.IsDNS1035Label(svm.Spec.Resource.Resource); len(errs) > 0 {
```

Comment:
Return `field.Required` error when resource is empty.

Reply:
Added